### PR TITLE
Setup basic windows compile

### PR DESCRIPTION
**Pull request by reverendbedford**
This PR makes compiling on Windows possible with NEML.

### CODE REVIEW

**Comment by reverendbedford** [comment]
Please review.  We need to merge this in now because it's painful to pull changes in from dev.

File: doc/sphinx/install/windows.rst [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+When installing Visual Studio include the include "Desktop development with C++"
+and "Python development" options.
+
+When installing the Intel compilers use the default selected packages.
+
+You will also need to install the `boost library <https://www.boost.org/users/download/>`_.  Installing boost is easy as it's a header library -- just unzip the archive to a convenient location.  In this example we use ``C:\Program Files\boost`` but you can put it wherever you want.
+
+Additionally, these directions assume you have the Windows command line
+version of `git <https://git-scm.com/download/win>`_ installed.
+
```
Comment (arovinelli) [inline doc/sphinx/install/windows.rst]: I guess but I'm not sure we need to get and install ninja, if so please add to the requirement list

Comment (reverendbedford) [reply doc/sphinx/install/windows.rst]: Nope, it comes with Visual Studio.

File: doc/sphinx/install/linux.rst [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+The main NEML library (in the :file:`lib/`) directory is all that needs to
+be linked to your software to call NEML material models.
+You only need to include the :file:`src/neml_interface.h` in order to
+load material models from XML datafile and use the resulting C++ object
+to call for the material response.
+
+The :file:`util/` directory contains example bindings of NEML into
+C++, C, and Fortran codes.  The CMake variable ``-D BUILD_UTILS=ON`` option
+compiles these example interfaces.
+During this option on requires a Fortran and C compiler.
```
Comment (chakra34) [inline doc/sphinx/install/linux.rst]: on->one?

File: doc/sphinx/install/windows.rst [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+=======
+
+These directions assume you are working on Windows 10 and you have installed
+`Microsoft Visual Studio <https://visualstudio.microsoft.com/>`_
+and the `Intel Fortran and C++ compilers <https://software.intel.com/en-us/parallel-studio-xe>`_.
+If you only want to use the python bindings then other compilers, including
+the Microsoft Visual C++ compiler, will likely work.  However,
+if you want to link to Abaqus or ANSYS through the :ref:`external interface <external>`
+then you will need a Fortran compiler, which seems to mean using the Intel
+Fortran compiler for most setups.  At that point you may was well use
```
Comment (chakra34) [inline doc/sphinx/install/windows.rst]: was->as

File: doc/sphinx/install/windows.rst [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+Intel C++ as well.
```
Comment (chakra34) [inline doc/sphinx/install/windows.rst]: no 'as well' in the end.

File: doc/sphinx/install/windows.rst [hunk 5]
```diff
@@ -0,0 +1,134 @@
+Windows
+=======
+
+These directions assume you are working on Windows 10 and you have installed
+`Microsoft Visual Studio <https://visualstudio.microsoft.com/>`_
+and the `Intel Fortran and C++ compilers <https://software.intel.com/en-us/parallel-studio-xe>`_.
+If you only want to use the python bindings then other compilers, including
+the Microsoft Visual C++ compiler, will likely work.  However,
+if you want to link to Abaqus or ANSYS through the :ref:`external interface <external>`
+then you will need a Fortran compiler, which seems to mean using the Intel
```
Comment (chakra34) [inline doc/sphinx/install/windows.rst]: seems to mean -> means ?

File: doc/sphinx/install/windows.rst [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+Open a shell with the Intel compilers setup by using the Start Menu to
+open "Compiler 19.0 Update 5 for Intel 64 Visual Studio 2019 Environment"
+(or similar, based on your version of the compilers).  This should open a
+Windows shell ready for use.
+
+Navigate to where you want to install NEML.  As an example
+
+.. code-block:: console
+
+  cd C:\Users\Mark\source\repos
```
Comment (chakra34) [inline doc/sphinx/install/windows.rst]: Mark -> $USER

File: doc/sphinx/install/windows.rst [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+search box in the Start Bar click "Environment Variables"
+
+.. image:: environment-variables.png
+
+then select "Path" in the "User variables" category, click "Edit"
+
+.. image:: path.png
+
+then click "New" and then enter the path to the ``lib`` directory in NEML, here
+this is ``C:\Users\Mark\source\repos\neml\lib``.  Accept all the open dialogs
```
Comment (chakra34) [inline doc/sphinx/install/windows.rst]: Mark->$USER

File: pybind11/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+    install(TARGETS pybind11 module embed
+            EXPORT "${PYBIND11_EXPORT_NAME}")
+    if(PYBIND11_MASTER_PROJECT)
+      install(EXPORT "${PYBIND11_EXPORT_NAME}"
+              NAMESPACE "${PROJECT_NAME}::"
+              DESTINATION ${PYBIND11_CMAKECONFIG_INSTALL_DIR})
+    endif()
+  endif()
+endif()
```
Comment (chakra34) [inline pybind11/CMakeLists.txt]: I am excluding checking CMake files, I know it works in linux centos and mac.

File: src/cp/slipharden.h [hunk 9]
```diff
@@ -20,50 +22,50 @@ namespace neml {
 class SlipRule; // Why would we need a forward declaration?
```
Comment (chakra34) [inline src/cp/slipharden.h]: this 'class' does not have the NEML_EXPORT ?

File: doc/sphinx/install/linux.rst (hunk 2 above)
Comment (reverendbedford) [reply doc/sphinx/install/linux.rst]: Actually During->Turning, but thanks.

File: doc/sphinx/install/windows.rst (hunk 3 above)
Comment (reverendbedford) [reply doc/sphinx/install/windows.rst]: Yes.

File: doc/sphinx/install/windows.rst (hunk 4 above)
Comment (reverendbedford) [reply doc/sphinx/install/windows.rst]: Yep.

File: doc/sphinx/install/windows.rst (hunk 5 above)
Comment (reverendbedford) [reply doc/sphinx/install/windows.rst]: This seems okay as is.

File: doc/sphinx/install/windows.rst (hunk 6 above)
Comment (reverendbedford) [reply doc/sphinx/install/windows.rst]: I compromised in the new commit.

File: doc/sphinx/install/windows.rst (hunk 7 above)
Comment (reverendbedford) [reply doc/sphinx/install/windows.rst]: Similarly compromised in the new commit.

File: pybind11/CMakeLists.txt (hunk 8 above)
Comment (reverendbedford) [reply pybind11/CMakeLists.txt]: K

File: src/cp/slipharden.h (hunk 9 above)
Comment (reverendbedford) [reply src/cp/slipharden.h]: That's a forward declaration, it doesn't need the export symbol.

**Comment by arovinelli** [comment]
After speaking with Mark I reluctantly agree to keep a hard copy of pybind in NEML just because was already there.

> Actually I really don't like the idea of hardcoding pybind. Please remove pybind from NEML and make a subrepo to an approved version

**Pull request merged**